### PR TITLE
chore(rpc): make `TransactionCompat::fill` stateful

### LINKED ISSUE #11164: Deposit receipt version in OP RPC transaction
### Describe the feature

What should we be setting the deposit receipt version field to in an `op_alloy_rpc_types::Transaction`? 

https://github.com/paradigmxyz/reth/blob/f606b70b7fd7ce01d7c501f86fc2d72658dd0306/crates/optimism/rpc/src/eth/transaction.rs#L115

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by emhane**
Ref https://github.com/paradigmxyz/reth/issues/11708
Closes https://github.com/paradigmxyz/reth/issues/11164

Makes `TransactionCompat::fill` take ref to self, to be able to add necessary data to `OpTxBuilder` to resolve the deposit receipt version for a transaction response

### CODE REVIEW

**Review by Rjected — CHANGES_REQUESTED** [review]
needs to instead implement `TransactionCompat` on the `OpEthApi`, so that we can fetch the receipt in the `fill` impl

File: crates/optimism/rpc/src/eth/transaction.rs [hunk 1]
```diff
@@ -88,30 +88,42 @@ where
...
             inner.gas_price = Some(signed_tx.max_fee_per_gas())
         }
 
+        let deposit_receipt_version = self
+            .inner
+            .provider()
+            .receipt_by_hash(hash)
+            .ok() // todo: change sig to return result
+            .flatten()
+            .and_then(|receipt| receipt.deposit_receipt_version);
```
Comment (Rjected) [inline crates/optimism/rpc/src/eth/transaction.rs:117]: let's do the todo here, maybe the error can be an associated type?

**Review by Rjected — CHANGES_REQUESTED** [review]
just one comment, otherwise this looks good to me

File: crates/optimism/rpc/src/eth/transaction.rs (hunk 1 above)
Comment (emhane) [reply crates/optimism/rpc/src/eth/transaction.rs:117]: it's quite a big change, effects signature of more rpc compat signatures, suggest it's done in a separate pr

Comment (Rjected) [reply crates/optimism/rpc/src/eth/transaction.rs:117]: alright, let's make an issue to track

Comment (emhane) [reply crates/optimism/rpc/src/eth/transaction.rs:117]: https://github.com/paradigmxyz/reth/issues/12020

**Review by Rjected — APPROVED** [review]
lgtm

**Pull request merged**